Задача 3. Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья
составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна
99%? (Провести двусторонний тест.)

In [15]:
import numpy as np

In [16]:
M = 200
sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
n = 10
a = (1 - 0.99) / 2

Так как дисперсия неизвестна , используем t – критерий.

In [17]:
mx = np.mean(sample)
t = (mx - M) / (np.std(sample, ddof=1) / np.sqrt(n))
print('t = ', t)
print('среднее значение по выборке =', mx)

t =  -1.0651074037450896
среднее значение по выборке = 198.5


Нулевая гипотеза утверждает, что средний вес пачки печенья составляет 200 г.
Альтернативная гипотеза утверждает, что средний вес пачки печенья отличен от 200 г.

для доверительной вероятности 99% и числа степеней свободы n - 1 = 9, табличное t = 2.821
т.к. среднее значение по выборке = 198.5 < 200, а t расчётное > t табличного / 2 (-1.0651 > -1.4105) принимаем нулевую гипотезу (средний вес пачки печенья составляет 200 г.)